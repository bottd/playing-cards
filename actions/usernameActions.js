export const actionTypes = {
  resetUsername: 'RESET_USERNAME',
  updateUsername: 'UPDATE_USERNAME',
}

export const resetUsername = () => ({
  type: actionTypes.resetUsername,
});

export const updateUsername = username => ({
  type: actionTypes.updateUsername,
  username,
});
