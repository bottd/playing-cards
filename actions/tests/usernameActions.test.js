import { actionTypes, resetUsername, updateUsername } from '../usernameActions';

describe('usernameActions', () => {
  describe('Action types', () => {
    it('Should match snapshot', () => {
      expect(actionTypes).toMatchSnapshot();
    });
  });

  describe('resetUsername', () => {
    it('Should return an action with correct type', () => {
      const expected = { type: 'RESET_USERNAME' };
      const result = resetUsername();
      expect(result).toEqual(expected);
    });
  });

  describe('updateUsername', () => {
    it('Should return an action with correct type', () => {
      const expected = 'UPDATE_USERNAME';
      const result = updateUsername('name');
      expect(result.type).toBe(expected);
    });
    it('Should return an action with the username', () => {
      const expected = 'name';
      const result = updateUsername(expected);
      expect(result.username).toBe(expected);
    });
    it('Should match snapshot', () => {
      const result = updateUsername('name');
      expect(result).toMatchSnapshot();
    });
  });
});
