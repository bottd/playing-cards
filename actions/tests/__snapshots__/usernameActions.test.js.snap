// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usernameActions Action types Should match snapshot 1`] = `
Object {
  "resetUsername": "RESET_USERNAME",
  "updateUsername": "UPDATE_USERNAME",
}
`;

exports[`usernameActions updateUsername Should match snapshot 1`] = `
Object {
  "type": "UPDATE_USERNAME",
  "username": "name",
}
`;
